doc/source/reference/c-api.iterator.rst

```diff
@@ -591,6 +619,15 @@ Construction and Destruction
             returns true from the corresponding element in the ARRAYMASK
             operand.
 
+        .. c:var:: NPY_ITER_OVERLAP_NOT_SAME
+
+            In the memory overlap checks done when ``NPY_ITER_COPY_IF_OVERLAP``
+            is specified, consider this array as overlapping even if it is
+            exactly the same as another array.
```

Comment:
I think this might be help clarify when the indexing is many to one in either array.